### DISCUSSION TITLE
グループライセンスでオフラインデバイスの登録が正常にできません。

### DISCUSSION THREAD

**Discussion by fit-st** [dashingsoft/pyarmor#1860] [Q&A]
グループライセンスでオフラインデバイスの登録が正常にできません。
License Typeがpyarmor-trialになり難読化ができません。

1. 初期登録

＞pyarmor reg -p SoftsensorOnlineTool pyarmor-keycode-6308.txt
INFO     Python 3.11.8
INFO     Pyarmor 8.5.10 (trial), 000000, non-profits
INFO     Platform windows.x86_64
INFO     prepare "pyarmor-keycode-6308.txt"
INFO     query key file from server
WARNING  this license has bound to product "soft sensor ONLINE"
WARNING  it can not be changed to "SoftsensorOnlineTool"
This license registration information will be

License Type    : pyarmor-group
License To      : XXXXXXXXX
License Product : soft sensor ONLINE

This license is about to be used for product "soft sensor ONLINE"

IMPORTANT: PRODUCT NAME CAN NOT BE CHANGED AFTER INITIAL REGISTRATION

Are you sure to continue? (yes/no) yes
INFO     process activation file "pyarmor-keycode-6308.txt"
INFO     send request to server
INFO     write registration file "pyarmor-regfile-6308.zip"
INFO     write group information
INFO     This group license has been activated sucessfully
INFO     Import Notes:

* Please backup regfile "pyarmor-regfile-6308.zip" carefully, and use this file for subsequent registration
* Do not use "pyarmor-keycode-6308.txt" again
* Please check `pyarmor reg` in Man page for how to register Pyarmor on offline device

2. グループデバイスファイルの生成

＞pyarmor reg -g 1 ./pyarmor-regfile-6308.zip

INFO     Python 3.11.8
INFO     Pyarmor 8.5.10 (trial), 000000, non-profits
INFO     Platform windows.x86_64
INFO     register device file ".pyarmor\group\pyarmor-group-device.1"
INFO     use group license "./pyarmor-regfile-6308.zip"
INFO     send request to server
INFO     write registration file "pyarmor-device-regfile-6308.1.zip"
INFO     please copy deivce regfile to offline device and run
INFO         pyarmor reg pyarmor-device-regfile-6308.1.zip
INFO     The device regfile has been generated successfully

3. オフラインデバイス regfile の生成

＞pyarmor reg -g 1 ./pyarmor-regfile-6308.zip
INFO     Python 3.11.8
INFO     Pyarmor 8.5.10 (trial), 000000, non-profits
INFO     Platform windows.x86_64
INFO     register device file ".pyarmor\group\pyarmor-group-device.1"
INFO     use group license "./pyarmor-regfile-6308.zip"
INFO     send request to server
INFO     write registration file "pyarmor-device-regfile-6308.1.zip"
INFO     please copy deivce regfile to offline device and run
INFO         pyarmor reg pyarmor-device-regfile-6308.1.zip
INFO     The device regfile has been generated successfully

4. オフラインデバイスにPyarmorを登録

＞pyarmor reg pyarmor-device-regfile-6308.1.zip
INFO     Python 3.11.8
INFO     Pyarmor 8.5.10 (trial), 000000, non-profits
INFO     Platform windows.x86_64
INFO     register "pyarmor-device-regfile-6308.1.zip"
INFO     machine id in group license: mb450c47849bb614a872c776c1060fa35
INFO     got machine id: mb450c47849bb614a872c776c1060fa35
INFO     this machine id matchs group license
INFO     This license registration information:

License Type    : pyarmor-trial
License No.     : pyarmor-vax-000000
License To      :
License Product : non-profits

BCC Mode        : No
RFT Mode        : No

Notes
* Can't obfuscate big script and mix str

**Comment by jondy** [comment]
Enable debug option in the offline device:
```
pyarmor -d reg pyarmor-device-regfile-6308.1.zip
```

Check the output log.

And try to unzip `pyarmor-device-regfile-6308.1.zip`, check the content of `tokens/mb450c47849bb614a872c776c1060fa35`

**Comment by jondy** [comment]
If `tokens/mb450c47849bb614a872c776c1060fa35` is empty, try to request it from server again
```
pyarmor reg -g 1 ./pyarmor-regfile-6308.zip
```

**Comment by fit-st** [comment]
Enable debug option in the offline device:

＞pyarmor -d reg pyarmor-device-regfile-6308.1.zip
INFO     Python 3.11.8
INFO     Pyarmor 8.5.10 (trial), 000000, non-profits
INFO     Platform windows.x86_64
DEBUG    native platform windows.amd64
DEBUG    home path: C:\Users\Fe\.pyarmor
INFO     register "pyarmor-device-regfile-6308.1.zip"
DEBUG    extracting license.lic
DEBUG    extracting .pyarmor_capsule.zip
INFO     machine id in group license: mb450c47849bb614a872c776c1060fa35
INFO     got machine id: mb450c47849bb614a872c776c1060fa35
INFO     this machine id matchs group license
DEBUG    extracting tokens/mb450c47849bb614a872c776c1060fa35
INFO     This license registration information:

License Type    : pyarmor-trial
License No.     : pyarmor-vax-000000
License To      :
License Product : non-profits

BCC Mode        : No
RFT Mode        : No

Notes
* Can't obfuscate big script and mix str


Check the output log.
And try to unzip pyarmor-device-regfile-6308.1.zip, check the content of tokens/mb450c47849bb614a872c776c1060fa35

F4sAAAEPAAAAHvi7AAAAAHB5YXJtb3ItdmF4LTAwNjMwOAAApZfUzn6enzgZVxuwdkIJZAAAAAAAAAAAAAAAACFtYjQ1MGM0Nzg0OWJiNjE0YTg3MmM3NzZjMTA2MGZhMzUXRlVKSSBFTEVDVFJJQyBDTy4sIExURC4Sc29mdCBzZW5zb3IgT05MSU5FBjE5MDEuMQ== PNKTQOV9ua6sV6DvzT603qToMR+e/JRRcbS9viL7otpiStSY1MvxKwjaJDaUXnXU9PDLJYbXQx5IDWBmSFxMrvAEUYo0jy8j9WIO0SEuuUYMb7238MTsS8dCLAANBwrOiCfbMnHOULjFIMIOJlX+MVU6PvdEgrrXVujQj4oe8evMSYrCHzI4iVFUTQMUQuXS5d401E7U1L6czC85/E7U+7MvwRmkX6ZRfoVRUxwEz5luOvCZxoVW3nX5d1nm2x2F6R3G3TeeidpX+QtOfZArClrV0eBjfdKj+NsHNuomFO73PD+ZNsfU/XsLi8WMBp5hzB8ZFrF/rxWxYfObbqbBtQ==

**Comment by jondy** [comment]
Is there file `C:\Users\Fe\.pyarmor\.license.group.token`? It may be hidden because it starts with `.`
And its content is same as tokens/mb450c47849bb614a872c776c1060fa35 or not?

**Comment by fit-st** [comment]
C:\Users\Fe\.pyarmor\.license.group.token

F4sAAAEPAAAAHvi7AAAAAHB5YXJtb3ItdmF4LTAwNjMwOAAApZfUzn6enzgZVxuwdkIJZAAAAAAAAAAAAAAAACFtYjQ1MGM0Nzg0OWJiNjE0YTg3MmM3NzZjMTA2MGZhMzUXRlVKSSBFTEVDVFJJQyBDTy4sIExURC4Sc29mdCBzZW5zb3IgT05MSU5FBjE5MDEuMQ== PNKTQOV9ua6sV6DvzT603qToMR+e/JRRcbS9viL7otpiStSY1MvxKwjaJDaUXnXU9PDLJYbXQx5IDWBmSFxMrvAEUYo0jy8j9WIO0SEuuUYMb7238MTsS8dCLAANBwrOiCfbMnHOULjFIMIOJlX+MVU6PvdEgrrXVujQj4oe8evMSYrCHzI4iVFUTQMUQuXS5d401E7U1L6czC85/E7U+7MvwRmkX6ZRfoVRUxwEz5luOvCZxoVW3nX5d1nm2x2F6R3G3TeeidpX+QtOfZArClrV0eBjfdKj+NsHNuomFO73PD+ZNsfU/XsLi8WMBp5hzB8ZFrF/rxWxYfObbqbBtQ==

ファイルはありました。
内容も同じです。

**Comment by jondy** [comment]
By the way, how about `pyarmor -v` in this offline machine?

**Comment by jondy** [comment]
Check if there is `C:\Users\Fe\.pyarmor\.license.token`, try to remove it if it exists.

**Comment by fit-st** [comment]
C:\Users\Fe\.pyarmor\.license.token を削除して pyarmor -v を行ったところ
License Typeがpyarmor-groupになりました。

ありがとうございました。
